packages/components/src/components/select/select.component.ts

```diff
@@ -0,0 +1,363 @@
+import type { TemplateResult } from 'lit';
+import { CSSResult, html, nothing } from 'lit';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import { KEYS } from '../../utils/keys';
+import { DataAriaLabelMixin } from '../../utils/mixins/DataAriaLabelMixin';
+import { DisabledMixin } from '../../utils/mixins/DisabledMixin';
+import { FormInternalsMixin } from '../../utils/mixins/FormInternalsMixin';
+import FormfieldWrapper from '../formfieldwrapper';
+import { DEFAULTS as FORMFIELD_DEFAULTS } from '../formfieldwrapper/formfieldwrapper.constants';
+import type { IconNames } from '../icon/icon.types';
+import { TAG_NAME as OPTION_GROUP_TAG_NAME } from '../optgroup/optgroup.constants';
+import { TAG_NAME as OPTION_TAG_NAME } from '../option/option.constants';
+import { POPOVER_PLACEMENT } from '../popover/popover.constants';
+import { TYPE, VALID_TEXT_TAGS } from '../text/text.constants';
+import { ARROW_ICON } from './select.constants';
+import styles from './select.styles';
+
+/**
+ * select component, represents a control that provides a menu of options.
+ *
+ * @dependency mdc-icon
+ * @dependency mdc-popover
+ * @dependency mdc-text
+ *
+ * @tagname mdc-select
+ *
+ * @slot default - This is a default/unnamed slot for options and/or option group.
+ *
+ * @event click - (React: onClick) This event is dispatched when the select is clicked.
+ * @event change - (React: onChange) This event is dispatched when the select is changed.
+ * @event keydown - (React: onKeyDown) This event is dispatched when a key is pressed down on the select.
+ * @event keyup - (React: onKeyUp) This event is dispatched when a key is released on the select.
+ * @event focus - (React: onFocus) This event is dispatched when the select receives focus.
+ */
+class Select extends FormInternalsMixin(DisabledMixin(DataAriaLabelMixin(FormfieldWrapper))) {
```

Comment:
```suggestion
class Select extends FormInternalsMixin(DataAriaLabelMixin(FormfieldWrapper)) {
```

DisabledMixin is already a part of FormfieldWrapper